### DISCUSSION TITLE
Detect Cheaters?

### DISCUSSION THREAD

**Discussion by rhunter-NTatC** [community/community#17735] [GitHub Education]
I was wondering what people used to detect cheaters?
Please post what you're doing.

I teach high school computer science and we use Eclipse and Java.
I encourage students to work collaboratively so there are times when code is very similar.

So, I'm just hoping to find something that could detect large sections of exact or near exact copies.
I can normally spot it when grading everything at once, but not when things are turned in late.  I'd need to comb through past submissions to find it.

It'd be REALLY cool if there were a place I could upload repositories or a list of their links and it'd just search for similarities.

Anyways, if you're using anything just post about it.

**Comment by pattacini** [comment]
Just wanted to link here some posts from the old community that are relevant to this hot topic:
- https://education.github.community/t/github-classroom-plagiarism-cheating-detection/11859
- https://education.github.community/t/how-to-anti-cheating-for-students-repo/51340
- https://education.github.community/t/github-classroom-timecode-cheating/48733

The cost we paid by moving to GitHub Discussions is that we somehow lose all that nice knowledge of the past.
It's still there, but barely browsed, I suppose.

Don't know if there's a way to make the connection easier.

cc @ericdrosado

**Reply by ericdrosado** [reply]
Unfortunately we haven't determined a way to integrate those threads into Discussions, but we do share a link to the old forum and additional resources [here](https://github.com/github-community#additional-resources). 

Thank you @pattacini for helping folks in the community discover these resources. 🙇

**Reply by rhunter-NTatC** [reply]
Thank you!
Looks like I've got a few things to try over summer break!

**Comment by tsaglam** [comment]
I know this is a bit late, but if anyone else needs a solution to detect source code plagiarism, feel free to try out https://github.com/jplag/JPlag. Here is the link to our [Demo](https://jplag.github.io/Demo/overview).

**Comment by VED-VIVEK-TALMALEY** [comment]
For detecting code plagiarism among Java assignments, consider these tools:

- MOSS (Measure of Software Similarity) - Stanford's academic tool that detects structural similarities despite variable name changes
- JPlag- Optimized for Java and can handle disguised copying
- CodeMatch- Compares code fragments with similarity scores
- GitHub Classroom- Has built-in similarity detection if you use GitHub
- PMD Copy-Paste Detector - Eclipse plugin for direct IDE integration

For batch comparison of repositories, MOSS and JPlag are best - they accept multiple submissions at once and generate similarity reports between file pairs.
